syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.ecnu.dbinter";
option java_outer_classname = "DbInterfaceProto";
option objc_class_prefix = "HLW";

package dbinter;

// The greeting service definition.
service DbInterface {

  rpc start(StartRequest) returns (StartReply) {}

  rpc get(GetRequest) returns (GetReply) {}

  rpc put(PutRequest) returns (PutReply) {}

  rpc commit(CommitRequest) returns (CommitReply) {}

  //rpc scan(ScanRequest) returns (ScanReply) {}

  //rpc SayHello (HelloRequest) returns (HelloReply) {}
}

message StartRequest {
  int64 transId = 1;
}

message StartReply {
  int64 transId = 1;
}

message GetRequest {
  int64 transId = 1;
  int64 tableId = 2;
  string key  = 3;
}

message GetReply { 
  string value = 1;
}

message PutRequest {
  int64 transId = 1;
  int64 tableId = 2;
  string key = 3;
  string value = 4;
}

message PutReply {
  int32 status = 1;
}

message CommitRequest { 
  int64 transId = 1;
}

message CommitReply {
  int32 status = 1;
}

